src/main.rs

```diff
@@ -1,3 +1,142 @@
-fn main() {
-    println!("Hello, World!");
+//! Test video and audio output by showing coloured frames on the framebuffer and playing back a
+//! siren over the default ALSA audio device.
+
+use alsa::pcm::{Access, Format, HwParams, State, PCM};
+use alsa::{Direction, ValueOr};
+use anyhow::{bail, Context};
+use framebuffer::{Framebuffer, KdMode};
+use std::{thread, time};
+
+const TTY: &str = "/dev/tty1";
+const FB: &str = "/dev/fb0";
+
+// how long a frame is displayed in seconds
+const FRAME_LENGTH: u64 = 1;
+// amount of time one audio sample should play
+const AUDIO_LENGTH: u64 = 1;
+
+fn frame_set_color(frame: &mut [u8], color: (u8, u8, u8)) {
+    for pixel in &mut frame.chunks_mut(4) {
+        pixel[0] = color.2;
+        pixel[1] = color.1;
+        pixel[2] = color.0;
+        pixel[3] = 0xff;
+    }
+}
+
+fn main() -> anyhow::Result<()> {
+    if let Some(a) = std::env::args().nth(1) {
+        if a == "--version" {
+            let name = env!("CARGO_PKG_NAME");
+            let version = env!("CARGO_PKG_VERSION");
+            eprintln!("{name}: {version}");
+            std::process::exit(0);
+        }
+    }
+
+    let frame_thread = thread::spawn(|| -> anyhow::Result<()> { frame() });
+    let siren_thread = thread::spawn(|| -> anyhow::Result<()> { siren() });
+
+    if let Ok(s) = frame_thread.join() {
```

Comment:
I disagree in this case: The variable is used in a one line long block and the context is clearly given by the call to `join`, which returns whatever value the joined thread returns after terminating. Artificially making the variable name longer will introduce noise.
To me this is similar enough to something like using `i` in a for loop that it's okay.

If it's not clear what the variable represents I'll make the change.

Reply:
To use i in a for loop is ok in most cases. But for me only if i is incremented only in single steps AND if it is not usfull to name it otherwise. And it is somthing it has a long "tradition" and is clear for everybody to read.
I do not really want to start a new tradition here. I don't even know for what s might be the short form. For status?

Reply:
I named the variable to carry more context this time, but I'm still against doing this in the future for this trivial case as it's not idiomatic for Rust to be this explicit when this much context is given already. A shorter form, which is not possible here because `join()` returns a type with specific constraints, would be `frame_thread.join()??`.

Reply:
One letter variables are a real pita. In every language. It is not necessary to blow the variable up to `frame_thread_join` but something what is better memorable then `s` would be nice. Something like `status` or `tstat` or what ever.